src/app/clusters/occupancy-sensor-server/OccupancySensingCluster.h

```diff
@@ -0,0 +1,126 @@
+
+        Config & WithFeatures(OccupancySensing::Feature featureMap)
+        {
+            mFeatureMap = featureMap;
+            return *this;
+        }
+
+        Config & WithHoldTime(uint16_t aHoldTime, const OccupancySensing::Structs::HoldTimeLimitsStruct::Type & aHoldTimeLimits,
+                              TimerDelegate & aTimerDelegate)
+        {
+            mHasHoldTime    = true;
+            mHoldTime       = aHoldTime;
+            mHoldTimeLimits = aHoldTimeLimits;
+            mTimerDelegate  = &aTimerDelegate;
+            return *this;
+        }
+
+        Config & WithDelegate(OccupancySensingDelegate * delegate)
+        {
+            mDelegate = delegate;
+            return *this;
+        }
+
+        Config & WithDeprecatedAttributes(bool showDeprecatedAttributes)
+        {
+            mShowDeprecatedAttributes = showDeprecatedAttributes;
+            return *this;
+        }
+
+        EndpointId mEndpointId;
+        BitMask<OccupancySensing::Feature> mFeatureMap                        = 0;
+        bool mHasHoldTime                                                     = false;
+        bool mShowDeprecatedAttributes                                        = true;
+        uint16_t mHoldTime                                                    = 10;
+        OccupancySensing::Structs::HoldTimeLimitsStruct::Type mHoldTimeLimits = { .holdTimeMin     = 1,
+                                                                                  .holdTimeMax     = 300,
+                                                                                  .holdTimeDefault = 10 };
+        TimerDelegate * mTimerDelegate                                        = nullptr;
+        OccupancySensingDelegate * mDelegate                                  = nullptr;
+    };
+
+    OccupancySensingCluster(const Config & config);
+    ~OccupancySensingCluster() = default;
+
+    CHIP_ERROR Startup(ServerClusterContext & context) override;
+    DataModel::ActionReturnStatus ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                AttributeValueEncoder & encoder) override;
+    DataModel::ActionReturnStatus WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                 AttributeValueDecoder & aDecoder) override;
+    CHIP_ERROR Attributes(const ConcreteClusterPath & path, ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder) override;
+
+    void TimerFired() override;
+
+    DataModel::ActionReturnStatus SetHoldTime(uint16_t holdTime);
+    void SetOccupancy(bool occupied);
+    bool IsOccupied() const;
+
+    uint16_t GetHoldTime() const;
+    const OccupancySensing::Structs::HoldTimeLimitsStruct::Type & GetHoldTimeLimits() const;
+
+private:
+    void DoSetOccupancy(bool occupied);
+
+    TimerDelegate * mTimerDelegate;
+    OccupancySensingDelegate * mDelegate;
+    BitMask<OccupancySensing::Feature> mFeatureMap;
```

Comment:
At least HoldTime is optional, and we have multiple attributes with conditions such as:
`[HoldTime & US]`, which means "May be implemented in the presence of the features..." so I believe we would need an OptionalAttributesSet here.